### PR TITLE
[SPARK-41005][COLLECT][FOLLOWUP] Remove JSON code path and use `RDD.collect` in Arrow code path

### PR DESCRIPTION
**Pull request by zhengruifeng**
### What changes were proposed in this pull request?
1, Remove JSON code path;
2, use RDD.collect in Arrow code path, since existing tests were already broken in Arrow code path;
3, reenable `test_fill_na`

### Why are the changes needed?
existing Arrow code path is still problematic and it fails and fallback to JSON code path, which change the output datatypes of `test_fill_na`

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
reenabled test and added UT

### CODE REVIEW

**Review by HyukjinKwon — APPROVED** [review]

File: connector/connect/src/main/scala/org/apache/spark/sql/connect/service/SparkConnectStreamHandler.scala [hunk 1]
```diff
@@ -57,13 +55,7 @@ class SparkConnectStreamHandler(responseObserver: StreamObserver[ExecutePlanResp
...
-      processAsArrowBatches(request.getClientId, dataframe)
-    } catch {
-      case e: Exception =>
-        logWarning(e.getMessage)
-        processAsJsonBatches(request.getClientId, dataframe)
-    }
+    processAsArrowBatches(request.getClientId, dataframe)
   }
 
   def processAsJsonBatches(clientId: String, dataframe: DataFrame): Unit = {
```
Comment (HyukjinKwon) [inline connector/connect/src/main/scala/org/apache/spark/sql/connect/service/SparkConnectStreamHandler.scala:61]: let's also remove this and protobuf definition

**Review by cloud-fan — APPROVED** [review]

**Comment by amaliujia** [comment]
Sorry for late comment but just one question:

does this implementation always send at least one partition to client even if there is empty result?

**Comment by zhengruifeng** [comment]
> Sorry for late comment but just one question:
> 
> does this implementation always send at least one partition to client even if there is empty result?


will send at least one batch.

**Comment by amaliujia** [comment]
@zhengruifeng thanks for this work! LGTM

**Pull request closed without merging**